synapseclient/models/table_components.py

```diff
@@ -482,6 +518,498 @@ def __repr__(self) -> str:
+
+    # Response attributes (filled after job completion)
+    job_id: Optional[str] = None
+    """The job ID returned from the async job"""
+
+    results_file_handle_id: Optional[str] = None
+    """The file handle ID of the results CSV file"""
+
+    table_id: Optional[str] = None
+    """The ID of the table that was queried"""
+
+    etag: Optional[str] = None
+    """The etag of the table"""
+
+    headers: Optional[List[SelectColumn]] = None
+    """The column headers from the query result"""
+
+    response_concrete_type: Optional[str] = None
+    """The concrete type of the response (usually DownloadFromTableResult)"""
+
+    def to_synapse_request(self) -> Dict[str, Any]:
+        """Convert to DownloadFromTableRequest format for async job submission."""
+        synapse_request = {
+            "concreteType": QUERY_TABLE_CSV_REQUEST,
+            "entityId": self.entity_id,
+            "csvTableDescriptor": {
+                "isFirstLineHeader": self.header,
+                "quoteCharacter": self.quote_character,
+                "escapeCharacter": self.escape_character,
+                "lineEnd": self.line_end,
+                "separator": self.separator,
+            },
+            "sql": self.sql,
+            "writeHeader": self.header,
+            "includeRowIdAndRowVersion": self.include_row_id_and_row_version,
+            "includeEntityEtag": True,
+        }
+        delete_none_keys(synapse_request)
+        return synapse_request
+
+    def fill_from_dict(self, synapse_response: Dict[str, Any]) -> "Self":
+        """Fill the job results from Synapse response."""
+        # Fill response attributes from DownloadFromTableResult
+        headers = None
+        headers_data = synapse_response.get("headers")
+        if headers_data and isinstance(headers_data, list):
+            headers = [SelectColumn.fill_from_dict(header) for header in headers_data]
+
+        self.job_id = synapse_response.get("jobId")
+        self.response_concrete_type = synapse_response.get("concreteType")
+        self.results_file_handle_id = synapse_response.get("resultsFileHandleId")
+        self.table_id = synapse_response.get("tableId")
+        self.etag = synapse_response.get("etag")
+        self.headers = headers
+
+        return self
+
+
+@dataclass
+class Query:
+    """
+    Represents a SQL query with optional parameters.
+
+    This result is modeled from: <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/table/Query.html>
+    """
+
+    sql: str
+    """The SQL query string"""
+
+    additional_filters: Optional[List[Dict[str, Any]]] = None
+    """Appends additional filters to the SQL query. These are applied before facets.
+    Filters within the list have an AND relationship. If a WHERE clause already exists
+    on the SQL query or facets are selected, it will also be ANDed with the query
+    generated by these additional filters."""
+    """TODO: create QueryFilter dataclass"""
```

Comment:
Do these TODO comments need to be completed now? Or could they be done later on?

Reply:
I can create additional tickets for these! I don't think they need to be completed now.